### ISSUE TITLE
UI not updated in real time when a new song is added

### ISSUE THREAD

**Issue by KaramveerSidhu** [navidrome/navidrome#1025]
### Description

When I add a new song/album to my music folder, the UI of navidrome is not updated in real-time. I have to reload the page to find my song in the list of albums.

### Ideal behavior

It would be great if the UI is updated in real-time.

### Screenshots

**Music Folder:** This is my music folder with a newly added song

![14-2](https://user-images.githubusercontent.com/51775341/114503600-37591f80-9c4b-11eb-8114-45b0c500baf2.PNG)

**Navidrome UI:** Here is the UI not updated in real-time

![14-1](https://user-images.githubusercontent.com/51775341/114503667-4cce4980-9c4b-11eb-981b-2ab191a96d4e.PNG)

It's only after reloading that a new album is displayed

![14-3](https://user-images.githubusercontent.com/51775341/114503786-71c2bc80-9c4b-11eb-86bc-319d204a1b00.PNG)

**Comment by pkspyder007** [comment]
@KaramveerSidhu You see there is a refresh button in the top-right header for refreshing the list.
For adding required behaviour we will need a socket connection so our server can send a refresh event in the UI.
And adding new songs is not a frequent operation and giving continuous resources for such task is not something I would want.

Let us see what others have to say about this.

**Comment by certuna** [comment]
Also, you can initiate a server rescan in the top right menu: Activity -> Quick Scan/Full Scan

**Comment by metalheim** [comment]
I'm also not sure if it's worth the hassle. I don't add songs that often and even when I do, I just wait 60s (or whatever SCAN_INTERVAL is set to) and hit the refresh button.

**Comment by deluan** [comment]
If we need to create a websocket connection and implement logic to detect if we actually need to reload the current view, it may not be worth the hassle.

BUT:

We already have a eventStream opened with the server, that replaces the websocket connection. We can create a new event on the server that is sent whenever the scan is completed, and in the UI we can just use the `useRefresh` hook to call refresh on the current view. We can even make it "smarter" by only sending the event when the DB is actually updated.

Not a priority IMO, as this is usually a rare situation (we don't add music all the time in our libraries, and we can always click the refresh view button), but a nice to have.

**Comment by metalheim** [comment]
@deluan I'm not sure if refreshing the current view forcefully is great. Imagine typing in the search bar and the UI just updates and you need to start typing again.
Maybe we could use that "DB updated" event to display a little `MuiBadge-badge` like that:

![image](https://user-images.githubusercontent.com/44159957/114585489-ff220300-9c83-11eb-97ab-219cd6d381e4.png)

**Comment by deluan** [comment]
@metalheim If we have to add a badge there, it may not be as straightforward to implement, and then I'll be forced to agree with you that it is not worth the hassle.

But refreshing the Ui should not cause a redraw:

https://user-images.githubusercontent.com/331353/114586218-6b840e80-9c52-11eb-8575-c09d96ba3bb1.mov

It only redraws the changed parts of the view. If there's no change, it does not redraw anything. The only exception is in the AlbumGrid, due to the way it is implemented, it always redraw. I hope we can fix it soon (maybe with the "New Album Grid" GSoC project), so refreshing it would be seamless as with the other views.

**Comment by metalheim** [comment]
Agree, i just tested refreshing during various user interactions and it doesn't break anything. So if we implement the feature, we should directly go for the refresh instead of just "displaying a badge"
